src/routes/webhooks/paddle.ts

```diff
@@ -66,6 +88,50 @@ const updateUserSubscription = async ({
   );
 };
 
+const logPaddleAnalyticsEvent = async (
+  data:
+    | SubscriptionUpdatedEvent
+    | SubscriptionCanceledEvent
+    | TransactionPaidEvent
+    | undefined,
+  eventName: AnalyticsEventName,
+) => {
+  if (!data) {
+    return;
+  }
+
+  const customData = data.data?.customData as { user_id: string };
+  const cycle = extractSubscriptionType(data.data?.items);
+  const cost = data.data?.items?.[0]?.price?.unitPrice?.amount;
+  const currency = data.data?.items?.[0]?.price?.unitPrice?.currencyCode;
+  const userId = customData?.user_id || data?.data.id || '';
+  const payment =
+    'payments' in data.data &&
+    data.data?.payments?.reduce((acc, item) => {
+      if (item.status === 'captured') {
+        acc = item?.methodDetails?.type || '';
+      }
+      return acc;
+    }, '');
+
+  const extra = {
+    cycle,
+    cost,
+    currency,
+    payment,
+  };
+
+  await sendAnalyticsEvent([
+    {
+      event_name: eventName,
+      event_timestamp: new Date(),
```

Comment:
Does the webhook provide timestamp? If so, let's use it here. If they provide it a unique id for the webhook payload, let's use it as event_id. This is to prevent dups